### PR TITLE
issue: 1575056 Fix deadlock during TCP connection establishment

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by liranoz12** [comment]
bot:retest

File: src/vma/sock/sockinfo_tcp.cpp [hunk 1]
```diff
@@ -3093,29 +3093,33 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
...
 	if (m_n_rx_pkt_ready_list_count)
-		goto noblock_nolock;
+		return true;
 
 	if (!p_poll_sn)
 		return false;
 
 	consider_rings_migration();
 
-	m_rx_ring_map_lock.lock();
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: Do we guarantee that `rx_add_ring_cb()` is not called during this function execution?
`rx_add_ring_cb()` can add new element into `m_rx_ring_map`

Comment (igor-ivanov) [reply src/vma/sock/sockinfo_tcp.cpp]: ?

Comment (liranoz12) [reply src/vma/sock/sockinfo_tcp.cpp]: rx_add_ring_cb() can be called from another thread - this it the main idea behind this deadlock.
But the map is still protected - see line 3118

**Comment by liranoz12** [comment]
@igor-ivanov please review

**Pull request merged**